internal/provider/http_helper.go

```diff
@@ -0,0 +1,33 @@
+package provider
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+func ExpectedResponse(response *http.Response, code int, diagnostics *diag.Diagnostics, err error) bool {
```

Comment:
Like the approach but I think we may be able to do better with the name of this function. What about `ValidateApiResponse` or `CheckApiResponseAndLog` (both ChatGPT generated but I think they're a bit more honest)

Reply:
I like `ValidateApiResponse` I didn't want to make the function name too long and that's a good suggestion :+1: